tests/unit/client.py

```diff
@@ -0,0 +1,659 @@
+        proc2.terminate()
+        cleanup(dir_names)
+        print("tearing down HIL setup")
+    request.addfinalizer(fin)
+
+
+@pytest.mark.usefixtures("create_setup")
+class Test_node:
+    """ Tests Node related client calls. """
+
+    def test_list_nodes_free(self):
+        result = C.node.list('free')
+        assert result == [u'node-06', u'node-07', u'node-08', u'node-09']
+
+    def test_list_nodes_all(self):
+        result = C.node.list('all')
+        assert result == [
+                u'node-01', u'node-02', u'node-03', u'node-04', u'node-05',
+                u'node-06', u'node-07', u'node-08', u'node-09'
+                ]
+
+    def test_show_node(self):
+        result = C.node.show('node-07')
+        assert result == {
+                u'metadata': {},
+                u'project': None,
+                u'nics': [
+                    {
+                        u'macaddr': u'aa:bb:cc:dd:ee:07',
+                        u'networks': {}, u'label': u'eth0'
+                        }
+                    ],
+                u'name': u'node-07'
+                }
+
+    def test_power_cycle(self):
+        result = C.node.power_cycle('node-07')
+        assert result is None
+
+    def test_power_off(self):
+        result = C.node.power_off('node-07')
+        assert result is None
+
+    def test_node_add_nic(self):
+        C.node.remove_nic('node-08', 'eth0')
+        result = C.node.add_nic('node-08', 'eth0', 'aa:bb:cc:dd:ee:ff')
+        assert result is None
+
+    def test_node_add_duplicate_nic(self):
+        C.node.remove_nic('node-08', 'eth0')
+        C.node.add_nic('node-08', 'eth0', 'aa:bb:cc:dd:ee:ff')
+        with pytest.raises(Exception):
+            C.node.add_nic('node-08', 'eth0', 'aa:bb:cc:dd:ee:ff')
+
+    def test_nosuch_node_add_nic(self):
+        with pytest.raises(Exception):
+            C.node.add_nic('abcd', 'eth0', 'aa:bb:cc:dd:ee:ff')
+
+    def test_remove_nic(self):
+        result = C.node.remove_nic('node-08', 'eth0')
+        assert result is None
+
+    def test_remove_duplicate_nic(self):
+        C.node.remove_nic('node-08', 'eth0')
+        with pytest.raises(Exception):
+            C.node.remove_nic('node-08', 'eth0')
+
+    def test_node_connect_network(self):
+        result = C.node.connect_network(
+                'node-01', 'eth0', 'net-01', 'vlan/native'
+                )
+        assert result is None
+
+    def test_node_start_console(self):
+        result = C.node.start_console('node-01')
+        assert result is None
+
+    def test_node_stop_console(self):
+        result = C.node.stop_console('node-01')
+        assert result is None
+
+
+# FIXME: I spent some time on this test. Looks like the pytest
+# framework kills the network server before it can detach network.
+# def test_node_detach_network(self):
+# C.node.connect_network('node-04', 'eth0', 'net-04', 'vlan/native')
+# result = C.node.detach_network('node-04', 'eth0', 'net-04')
+# assert result is None
```

Comment:
Same thing as above; don't leave code commented out. Can you explain to me what was happening here?